app/models/secret_storage.rb

```diff
@@ -50,6 +50,49 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    VAULT_SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(vault_path(key))
+      return false if result.nil?
+      result.data[:vault]
+    end
+
+    def self.write(key, data)
+      Vault.logical.write(vault_path(key), vault: data)
+    end
+
+    def self.delete(key)
+      Vault.logical.delete(vault_path(key))
+    end
+
+    def self.keys()
+      Vault.logical.list(VAULT_SECRET_BACKEND).map! { |key| encode_path(key) }
+    end
+
+    private
+
+    def self.vault_path(key)
+      VAULT_SECRET_BACKEND + encode_path(key, :encode)
+    end
+
+    def self.encode_path(string = nil, direction = :decode)
```

Comment:
don't need the nil default and return guard ... don't need the default direction ...